### ISSUE TITLE
mempool: Several issues with GetVoteHashesForBlock

### ISSUE THREAD

**Issue by davecgh** [decred/dcrd#481]
- Per Effective Go, the function should not start with `Get`.  It should be `VoteHashesForBlock`.
- The function should not be returning errors if votes for the requested block hash don't exist, rather it should simply return a `nil` array.  It should be up to the caller to determine if the lack of votes for a given block is an error or not.
- The function should not be returning an error in the case of an unset vote hash.  First, such transactions should never be allowed into the mempool to begin with as they are invalid, and second, even if they were valid, it would be up to the caller to make the decision whether or not the hashes for a vote are acceptable to it or not.

In short, the entire function shouldn't return errors at all.  The caller should request the votes that exist for a given block and unconditionally be provided them.  The function should literally be nothing more than grabbing a mutex and returning `mp.votes[hash]`.

EDIT: I should also note that part of changing this must entail ensuring the callers are properly checking for error conditions as they should already be doing.
